crates/api/src/lib.rs

```diff
@@ -69,6 +71,30 @@ pub(crate) fn check_report_reason(reason: &str, local_site: &LocalSite) -> Resul
   }
 }
 
+pub fn read_auth_token(req: &HttpRequest) -> Result<Option<String>, LemmyError> {
+  // Try reading jwt from auth header
+  if let Ok(header) = Authorization::<Bearer>::parse(req) {
+    Ok(Some(header.as_ref().token().to_string()))
+  }
+  // If that fails, try auth cookie. Dont use the `jwt` cookie from lemmy-ui because
+  // its not http-only.
```

Comment:
I don't understand this separation between the jwt and the `auth` cookie. Why are we adding a second cookie with the same content just to have one be client-readable and one be http-only? Why is the jwt token returned in the response body at all?

Reply:
I was thinking to use a different cookie name because the `jwt` cookie sent by lemmy-ui has `httponly=false` so it would fail the checks here. But its probably possible to make it work, so I changed the cookie name to `jwt` and provided a clear error message.

Reply:
Great. That will need testing if it causes any weird behaviour in lemmy-ui because that currently manually also set's the same cookie.